public/app/features/explore/ExploreActions.tsx

```diff
@@ -65,6 +70,25 @@ export const ExploreActions = () => {
         });
       }
     } else {
+      // command palette doesn't know what pane we're in, only show option if not split and no datasource is mixed
```

Comment:
why only if not splitted? in the UI i can still open the correlation editor in split mode (even if the right pane is using mixed datasource)
this should probably be consistent with that

Reply:
You're right, this doesn't make sense. I changed it so it should do the following

If you change the datasource on either side and there is a dirty state, prompt for save/discard/cancel. Discarding will change the datasource after.

If there is no dirty state, remove the helper data. It will still be in split mode.

Reply:
My thinking here is wanting to be sure the user would perform an action they'd expect. 

We want to be consistent that when you open the correlation editor, you are starting with the source datasource. Source datasource on the left pane, target on the right. Starting into this mode should start with the source, so the left pane.

On the toolbar, the buttons are tied to the pane. The command palette is an overlay over both panes. If the screen is split, and you want to open it in correlation editor mode, there is no question on what you're doing if you press the button tied to the left pane - you are starting to build a correlation with the source datasource. If you initiate it screen wide, you might think you can start a correlation from the right pane and that's not what will happen.

This is all a little involved and I might be in my own head about it, but I'd like to keep this limitation for now and explore opening it up if we hear anything about wanting to change it.